src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -42,12 +77,84 @@ public async Task ExecuteAsync(ExportJobRecord exportJobRecord, WeakETag weakETa
 
             try
             {
-                // TODO: Implement the actual export logic here.
+                // Get destination type from secret store.
+                DestinationInfo destinationInfo = await GetDestinationInfo(cancellationToken);
+
+                // Connect to the destination using appropriate client.
+                _exportDestinationClient = _exportDestinationClientFactory.Create(destinationInfo.DestinationType);
+
+                await _exportDestinationClient.ConnectAsync(destinationInfo.DestinationConnectionString, cancellationToken);
+
+                // TODO: For now, always restart from the beginning. We will support resume in another work item.
+                exportJobRecord.Progress = new ExportJobProgress(continuationToken: null, page: 0);
+
+                ExportJobProgress progress = exportJobRecord.Progress;
+
+                // Current page will be used to organize a set of search results into a group so that they can be committed together.
+                uint currentPage = progress.Page;
+
+                // The first item is placeholder for continuation token so that it can be updated efficiently later.
+                var queryParameters = new Tuple<string, string>[]
+                {
+                    null,
+                    Tuple.Create(KnownQueryParameterNames.Count, _exportJobConfiguration.MaximumNumberOfResourcesPerQuery.ToString(CultureInfo.InvariantCulture)),
+                    Tuple.Create(KnownQueryParameterNames.LastUpdated, $"le{exportJobRecord.QueuedTime.ToString("o", CultureInfo.InvariantCulture)}"),
+                };
+
+                // Process the export if:
+                // 1. There is continuation token, which means there is more resource to be exported.
+                // 2. There is no continuation token but the page is 0, which means it's the initial export.
+                while (progress.ContinuationToken != null || (progress.ContinuationToken == null && progress.Page == 0))
```

Comment:
nit: can be simplified to `(progress.ContinuationToken != null || progress.Page == 0)`